nexus/src/nexus.rs

```diff
@@ -2307,6 +2316,31 @@ impl Nexus {
         }
     }
 
+    /*
+     * Timeseries
+     */
+
+    /**
+     * List existing timeseries schema.
+     */
+    pub async fn timeseries_schema_list(
+        &self,
+        pag_params: &TimeseriesSchemaPaginationParams,
+        limit: NonZeroU32,
```

Comment:
I found it a little surprising that the `limit` shows up separately here because `PaginationParams` already has one.  I can see why we got here though: you can really only access that limit via `rqctx` because it may have a default or cap applied, and you don't want to pass the dropshot RequestContext all the way down.  Elsewhere, we made an internal representation of pagination params -- that's `DataPageParams`.  I can see it might not make sense to bother with that here.

Reply:
That's all accurate.

The implementation of pagination is a bit scattered now, as the `Client` is isn't really a Dropshot server. Nexus is the only place we have the `RequestContext`, which is the only way to get the limit from inside the `PaginationParams` type. This was the cleanest separation I could find at the time.

And yeah, I was trying to avoid adding too much boilerplate. I considered using the `DataPageParams`, but that currently works with `Name`, which is different from a `TimeseriesName`. I also didn't want to add a dep on `omicron-common` at this point.

So short answer, I don't want to bother now, but we can revisit if it becomes useful.